dom.go

```diff
@@ -505,9 +507,10 @@ func (h *HTML) reconcileChildren(prev *HTML) (pendingMounts []Mounter) {
 		}
 	}
 
-	// If dealing with keyed siblings, remove children by key rather than
-	// position.
-	if prev.hasKeyedChildren && h.hasKeyedChildren {
+	// If dealing with keyed siblings, remove all prev.keyedChildren which are
+	// leftovers / ones we did not find a match for above.
+	if len(prev.keyedChildren) > 0 && hasKeyedChildren {
```

Comment:
Actually @pdf the only case where we care about `len(prev.keyedChildren)` is here. And because of the `> 0` it's always guaranteed that we have at least one for the case we care about. For example, if all `prev.keyedChildren` were deleted, this case does not need to run. In contrast, if one exists, this case will run.

Reply:
But it does need to match, so that we don't proceed with the rest of the logic in this function.

Reply:
Right... now that I've reverted the `else` block I added here you are correct.

This seems like a good thing to handle with an `else` block though, do you disagree?

Reply:
5bc9b35fc24af320e100399d9fbfd47d3fb3fece should solve this